linera-execution/src/wasm/runtime_actor/responses.rs

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Types related to sending responses from an actor after handling requests.
+
+use futures::channel::oneshot;
+use std::{
+    mem,
+    sync::{Arc, Condvar, Mutex},
+};
+use thiserror::Error;
+
+/// The sender endpoint for a synchronous single-message channel.
+pub struct SyncResponseSender<T: Send>(Arc<SyncResponse<T>>);
```

Comment:
If we're going to create our one own sync one-shot channel, we'll need a few unit tests. Also, have you looked at the crate [`oneshot`](https://docs.rs/oneshot/latest/oneshot/)?

Reply:
Nice find! I was able to replace all the custom senders with it, and it made the code simpler :tada: Thanks!